tests/test_config.py

```diff
@@ -0,0 +1,77 @@
+import unittest
+from unittest import mock
+from unittest.mock import patch, mock_open
+from textwrap import dedent
+import os
+from codecarbon.core.config import (
+    clean_env_key,
+    parse_env_config,
+    get_hierarchical_config,
+)
+from pathlib import Path
+
+
+class TestConfig(unittest.TestCase):
+    def test_clean_env_key(self):
+        for key in [1, None, 0.2, [], set()]:
+            with self.assertRaises(AssertionError):
+                clean_env_key(key)
+        for (key, target) in [
+            ("", ""),
+            ("USER", "user"),
+            ("CODECARBON_TEST", "test"),
+            ("CODECARBON_TEST_VALUE", "test_value"),
+            ("CODECARBON_TEST_1", "test_1"),
+            ("CODECARBON_1", "1"),
+        ]:
+            self.assertEqual(clean_env_key(key), target)
+
+    @mock.patch.dict(
+        os.environ,
+        {
+            "USER": "yes",
+            "CODECARBON_TEST": "test-VALUE",
+            "CODECARBON_TEST_KEY": "this_other_value",
+        },
+    )
+    def test_parse_env_config(self):
+        self.assertDictEqual(
+            parse_env_config(),
+            {"codecarbon": {"test": "test-VALUE", "test_key": "this_other_value"}},
+        )
+
+    def custom_mock_open(self, *args, **kwargs):
```

Comment:
Custom mock to selectively return the right `read_data` as a function of the `path` so that the two calls to `ConfigParser().read(...)` are patched in `get_hierarchical_config`